contracts/payment/MCPayment.sol

```diff
@@ -18,6 +19,12 @@ contract MCPayment is Ownable2StepUpgradeable, EIP712Upgradeable {
             "Iden3PaymentRailsRequestV1(address recipient,uint256 amount,uint256 expirationDate,uint256 nonce,bytes metadata)"
         );
 
+    bytes32 public constant ERC_20_PAYMENT_DATA_TYPE_HASH =
+        keccak256(
+            // solhint-disable-next-line max-line-length
+            "Iden3PaymentRailsERC20RequestV1(address tokenAddress,address recipient,uint256 amount,uint256 expirationDate,uint256 nonce,bytes metadata)"
+        );
+
```

Comment:
Maybe we could have only 1 PAYMENT_DATA_TYPE adding the `tokenAddress` and check that this `tokenAddress` is 0x0 for native token?

Reply:
Not sure that validating in each native token payment is better than having one more constant. @AndriianChestnykh what do you think ?

Reply:
@volodymyr-basiuk I agree. Different reasons to change
You should do `erc20Payment` payable then and check msg.value == 0 or !=0 in different cases, can't securely resent ETH, but may resend tokens to issuer... what if you add other than erc20 asset payment, etc..
However two function names `pay` and `erc20Payment` could've been more consistent to each other.

Reply:
I’m not in favor of using a single hash, as these hashes represent two different payment types at the protocol level. We previously discussed reusing the Iden3PaymentRailsRequestV1 type for ERC-20 payments and, if the payment was made on the native currency, passing 0x for the tokenAddress. However, this approach only adds confusion, so we decided to provide a separate type - Iden3PaymentRailsERC20RequestV1.

Reply:
two different data domains for different protocol types is made decision for now.